airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/xmin/PostgresXminHandler.java

```diff
@@ -138,24 +138,75 @@ private PreparedStatement createXminQueryStatement(
           quoteString);
 
       final String wrappedColumnNames = RelationalDbQueryUtils.enquoteIdentifierList(columnNames, quoteString);
+
+      // Get the xmin status associated with the previous run
+      final XminStatus previousRunXminStatus = xminStateManager.getXminStatus(airbyteStream);
+      final PreparedStatement xminPreparedStatement =
+          getXminPreparedStatement(connection, wrappedColumnNames, fullTableName, previousRunXminStatus, currentXminStatus);
+      LOGGER.info("Executing query for table {}: {}", tableName, xminPreparedStatement);
+      return xminPreparedStatement;
+    } catch (final SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private PreparedStatement getXminPreparedStatement(final Connection connection,
+      final String wrappedColumnNames, final String fullTableName, final XminStatus prevRunXminStatus, final
+      XminStatus currentXminStatus) throws SQLException {
+
+    // If the source Postgres DB has undergone multiple wraparound events between syncs, we have no option but to perform
+    // a full sync.
+    if (shouldPerformFullSync(prevRunXminStatus, currentXminStatus)) {
+      final String sql = String.format("SELECT %s FROM %s", wrappedColumnNames, fullTableName);
+      return connection.prepareStatement(sql);
+    } else if (isSingleWraparound(prevRunXminStatus, currentXminStatus)) {
       // The xmin state that we save represents the lowest XID that is still in progress. To make sure we
-      // don't miss
-      // data associated with the current transaction, we have to issue an >=
+      // don't miss data associated with the current transaction, we have to issue an >=. Because of the wraparound, the changes prior to the
+      // end xmin xid value must also be captured.
+      final String sql = String.format("SELECT %s FROM %s WHERE xmin::text::bigint >= ? OR xmin::text::bigint < ?",
```

Comment:
What is the expectation in terms of large table?
Are two logical > < going to take twice as much time?

Reply:
No, it still requires just a full table scan. The query optimizer ends up checking for a single filter condition.